drivers/flash/Kconfig.gd32

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022 BrainCo Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FLASH_GD32
+	bool "GigaDevice GD32 flash driver"
+	default y
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
+	select MPU_ALLOW_FLASH_WRITE if ARM_MPU
+	depends on (SOC_FAMILY_GD32 || SOC_SERIES_GD32VF103)
```

Comment:
My suggestion is:

Add `GD32_HAS_FLASH_DRIVER` at modules/hal_gigadevice

Check by `GD32_HAS_FLASH_DRIVER` instead SoC Family or Series.
```diff
- depends on (SOC_FAMILY_GD32 || SOC_SERIES_GD32VF103)
+ depends on GD32_HAS_FLASH_DRIVER
```

Than add `GD32_HAS_FLASH_DRIVER` at SoC https://github.com/zephyrproject-rtos/zephyr/blob/bc72fb67b1c81670a087f766ef38c3517da10366/soc/arm/gigadevice/gd32f4xx/Kconfig.series#L11-L12

This way flash driver will be only available for those SoC that have a driver.

Reply:
done.